build.gradle

```diff
@@ -47,30 +49,23 @@ allprojects {
 }
 
 subprojects {
-  apply plugin: "org.jlleitschuh.gradle.ktlint"
+  apply plugin: "io.gitlab.arturbosch.detekt"
 
   repositories {
     google()
     mavenCentral()
   }
 
-  ktlint {
-    version = "0.43.2"
-    enableExperimentalRules = false
-    reporters {
-      reporter "checkstyle"
-    }
+  detekt {
+    config = files("$rootDir/.detekt.yml", "$rootDir/.detekt-compose.yml")
+    buildUponDefaultConfig = true
+    autoCorrect = true
```

Comment:
Is this going to automatically fix issues when running detekt? how's that working on CI - particularly with the snapshot testing checking for any files changing when recording snapshots.
also in general i prefer explicitly running a fix command - that way it's easier to look at the issues and potentially manually fixing something (ie automatic line breaks/formatting can sometimes cause very odd formatting

Reply:
It will only fix the issues on the 2nd run, so CI is fine. Double-running to fix it is an approach used widely in the industry and it's generally safe. And faster, more convinient – if your commit fails, just run it again, instead of opening terminal, running the command and try to commit again (or trying to manually resolve the issues from the commit log).

Reply:
Are you sure this is what it's doing? Afaik it'll fix the errors, but fail the task to make you aware of it. Next time you run it it succeeds because the errors were fixed in the previous run. 
Maybe instead of setting `autoCorrect` here we can change the git hook to use the `--auto-correct` property if we want the behaviour of auto fixing for that, but keep CI separate. That way also if somebody wants to just run detekt without auto correcting that is still possible.

Reply:
Done